addons/udes_delivery_control/models/picking_extra_fields.py

```diff
@@ -0,0 +1,146 @@
+    u_backload_cover_count = fields.Integer(
+        "Cover Count", help="Number of covers backloaded", default=0,
+    )
+    u_backload_reject_count = fields.Integer(
+        "Reject Count",
+        help="Number of pallets/stillages/covers rejected and not backloaded",
+        default=0,
+    )
+    u_backload_start_date = fields.Datetime("Start Date", help="Backload Start Date")
+    u_backload_end_date = fields.Datetime("End Date", help="Backload End Date")
+    u_backload_time_taken = fields.Float(
+        "Time Taken HH:MM",
+        compute="_compute_u_backload_time_taken",
+        store=True,
+        readonly=True,
+        help="The amount of time taken to complete back loading in HH:MM format",
+    )
+
+    _sql_constraints = [
+        ("picking_id_uniq", "unique(picking_id)", "Only one record is allowed per picking"),
+    ]
+
+    @api.multi
+    @api.depends("u_vehicle_arrival_date")
+    def _compute_u_week(self):
+        """ Compute method of u_week field """
+        for record in self:
+            record.u_week = 0
+            if record.u_vehicle_arrival_date:
+                record.u_week = fields.Datetime.from_string(
+                    record.u_vehicle_arrival_date
+                ).isocalendar()[1]
+
+    @api.multi
+    @api.depends("u_unloading_start_date", "u_unloading_end_date")
+    def _compute_u_time_taken(self):
+        """ Compute u_time_taken field from unloading start and end date """
+        for record in self:
+            record.u_time_taken = 0
+            if record.u_unloading_start_date and record.u_unloading_end_date:
+                record.u_time_taken = date_diff(
+                    record.u_unloading_start_date, record.u_unloading_end_date
+                )
+
+    @api.multi
+    @api.depends("u_backload_start_date", "u_backload_start_date")
+    def _compute_u_backload_time_taken(self):
+        """ Compute u_backload_time_taken field """
+        for record in self:
+            record.u_backload_time_taken = 0
+            if record.u_backload_start_date and record.u_backload_end_date:
+                record.u_backload_time_taken = date_diff(
+                    record.u_backload_start_date, record.u_backload_end_date
+                )
+
+    @api.constrains("u_unloading_start_date", "u_unloading_end_date")
+    def _check_unloading_dates(self):
+        msg = _("Unloading start time cannot be greater than end time")
+        validate_dates(self.u_unloading_start_date, self.u_unloading_end_date, msg)
+
+    @api.constrains("u_backload_start_date", "u_backload_end_date")
+    def _check_backload_dates(self):
+        msg = _("Backloading start time cannot be greater than end time")
+        validate_dates(self.u_backload_start_date, self.u_backload_end_date, msg)
+
+    @api.model
+    def create(self, values):
+        res = super(StockPickingDeliveryControl, self).create(values)
+        res._update_picking_data()
+        return res
+
+    def _update_picking_data(self):
```

Comment:
you don't need this if you have the back ref

Reply:
We might need picking_id search in database for reporting purpose. Having picking_id makes it easier?

Reply:
If you have the reverse connection you can use the odoo magic to populate the values for you.
```(0, 0, values)
    adds a new record created from the provided value dict.```

Reply:
done